### PR TITLE
[Sample] Extract to EML file

### PR DESCRIPTION
**Pull request by inter6**
1. Add maven-compiler-plugin
1) set 1.6 Java Version
2) set UTF-8 Project encoding

2. Test Code - Extracts the mail(in the IPF.Imap folder) to EML file.
1) Travel Folder.
2) Create MimeMessage from PSTMessage.
3) Write to EML file.

### CODE REVIEW

**Comment by andreak** [comment]
Hi.
I'd love to see this PR merged as it provides the ability to save a PSTMessage to an EML-file.

What's holding up the merge?

Thanks.

**Comment by rjohnsondev** [comment]
@andreak 

Firstly this PR incorrectly includes a bunch of commits from the 2013 ost support branch and other changes that should not be included.

Secondly, I'm unsure of the intent of this PR as this seems to be example code for writing to a EML file, however it's implemented as a test.  If the intent is to give an example usage of the library, I'd be happy to see it written as a usable executable and included in /src/main/java/example, or simply have the readme point to a complete example project that utilises the library on the author's github.

In a test context, the code references a pst with a hard-coded path that exists only on author's machine, so it is not useful for anyone but the author (https://github.com/rjohnsondev/java-libpst/pull/25/files#diff-ebe1217e8c26757516c818e4ce718135R42).

Basically, this is not something that is mergeable.  If you would like to take this code and create a useful example application from it, I'd be happy to include a link to it from the readme or docs!

**Comment by andreak** [comment]
@rjohnsondev 
Thanks for reply!
It would be nice to have a working example of how to write emails (PSTMessage) as EML-files.
If this is possible and I've overlooked it, please point me to some code showing how to do it.

Thanks.

**Comment by rjohnsondev** [comment]
Despite my objections to this PR, the code (https://github.com/rjohnsondev/java-libpst/pull/25/files#diff-ebe1217e8c26757516c818e4ce718135R93) looks roughly to do the right thing for what you are after.

As I said, if you take this code and create a useful example application from it, let me know.

**Comment by andreak** [comment]
That code builds a MimeMessage "manually" and writes it to an outputstream. The crafted MimeMessage is built with a static structure (the root-part is always multipart/mixed for example), not following the original email's structure, it seems.
If I open a PST-file in Outlook and save an email to disk, the generated eml-file from libpst should always be equal to it.

**Comment by rjohnsondev** [comment]
@andreak PSTs do not store mime messages as such, emails are stored as a collection of keys and values.  Manually creating the mimemessage based on the properties (like getTransportMessageHeaders) is the only way to achieve this AFAIK.

**Comment by andreak** [comment]
Ok, thanks for the clarification.
